scripts/collect_model_urls.py

```diff
@@ -0,0 +1,22 @@
+import pathlib
+import re
+import sys
+
+MODEL_URL_PATTERN = re.compile(r"https://download.pytorch.org/models/.*?[.]pth")
+
+
+def main(root):
+    model_urls = set()
+    for path in pathlib.Path(root).glob("**/*"):
+        if path.name.startswith("_") or not path.suffix == ".py":
+            continue
+
+        with open(path, "r") as file:
+            for line in file:
+                model_urls.update(MODEL_URL_PATTERN.findall(line))
+
+    print("\n".join(sorted(model_urls)))
```

Comment:
This approach is a bit hacky, though admittedly you can't do much else prior compiling torchvision. If TorchVision was compiled, we could rely on the upcoming registration mechanism to get all available models and weights and then fetch their URLs. Since this is not possible for speed reasons, we might be forced to do something like that. The good thing is even if this fails, we will download the weights properly one-by-one later.

Reply:
Seeing how fast the download is, we could also do it in the foreground after the installation. Still much faster than downloading sequentially during the tests. Both variants are fine by me. You pick.

Reply:
13 seconds sounds pretty good to me. It's your call. I don't mind either way.